src/containerapp/azext_containerapp/custom.py

```diff
@@ -4332,3 +4339,258 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
         return r
     except Exception as e:
         handle_raw_exception(e)
+
+
+def patch_list(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.error("Please install or start Docker and try again.")
+        return
+    pack_exec_path = get_pack_exec_path()
+    if pack_exec_path is None:
+        return
+    print("\rListing container apps...", end="", flush=True)
+    ca_list = list_containerapp(cmd, resource_group_name, managed_env)
+    imgs = []
+    if ca_list:
+        for ca in ca_list:
+            resource_group_name = re.search('/subscriptions/[^/]+/resourceGroups/([^/]+)/', ca["id"]).group(1)
```

Comment:
resourcegroups should also work. Could you ignore case for resource id?

Reply:
consider using parse_resource_id():

e.g.
```
        id_parts = parse_resource_id(template_spec)
        resource_group_name = id_parts.get('resource_group')
        name = id_parts.get('name')
```

Reply:
Thanks for the suggestion! Moved to `parse_resource_id` by @harryli0108.